Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -278,7 +278,7 @@ export class ResourceGroup {
     private createDeploymentName(): string {
         var name: string;
         if (this.taskParameters.templateLocation == "Linked artifact") {
-            name = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), this.taskParameters.csmFile)[0];
+            name = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), this.taskParameters.csmFile.replace(/[\[]/g, '$&[]'))[0];
```

Comment:
the replacement can be done while populating `this.taskParameters.csmFile` at the beginning of the task. This way you do not need to replace at multiple places.

Reply:
We might do non globbing operations as well on these file paths so its not a good idea to modify them at starting of the tasks. Thoughts ?

Reply:
My intention is that we do not repeat the same logic multiple times. Another option is to add a common function call from all these places.

Reply:
Done